wp_all_import_functions.php

```diff
@@ -1,6 +1,103 @@
+             $logs[] = implode(' ', $info);
+        }
+
+        return new WP_REST_Response(['status' => 'success', 'logs' => $logs, 'data' => $data], 200);
+    } else {
+        return new WP_REST_Response(['status' => 'error', 'message' => 'Missing params'], 500);
+    }
+}
+
+
+function update_post_links( $postid, $spostid, $mime_type= 'pdf'):array {
+    $content_post = get_post( $postid );
+    $content = $content_post->post_content;
+    $content = apply_filters( 'the_content', $content );
+
+    if ($mime_type === 'pdf') {
+        $attachments = get_attached_media( 'application/pdf', $spostid );
+        preg_match_all( '@href="([^"]+\.pdf|PDF)"@' , $content, $match_pdf );
+        $media_files = array_pop( $match_pdf );
+
+        // check for single quote pdf files.
+        preg_match_all( "@href='([^']+\.pdf|PDF)'@" , $content, $single_quote_match_pdf );
+        $single_quote_media = array_pop( $single_quote_match_pdf );
+        $media_files = array_merge($media_files, $single_quote_media);
+    }
+
+    if ($mime_type === 'img') {
+        $attachments = get_attached_media( 'image', $spostid );
+        preg_match_all( '@src="([^"]+)"@' , $content, $match_img );
+        $media_files = array_pop( $match_img );
+        // check for single quote images
+        preg_match_all( "@src='([^']+)'@" , $content, $single_quote_match_img );
+        $single_quote_media = array_pop( $single_quote_match_img );
+        $media_files = array_merge($media_files, $single_quote_media);
+    }
+
+    $update_status = 0;
+
+    // Fix broken url in Post content.
+    foreach ( $media_files as $media_file ) {
+        // If media file belongs to P4 stateless bucket, skip it.
+        if (strpos($media_file,"planet4-usa-stateless")) {
+            continue;
+        }
+
+        $basename = basename( $media_file );
+        $basename = str_replace(' ', '-', $basename);
+        $basename = str_replace('%20', '-', $basename);
+        foreach ( $attachments as $attachment ) {
+            $guid = $attachment->guid;
+
+            /*// If same image/pdf imported twice, the stateless append -1 in file name.
+            if (preg_match( '/\-1\.jpg$/i', $guid )) {
+                $guid = str_replace( '-1.jpg', '.jpg', $guid );
+            }
+            if (preg_match( '/\-1\.pdf$/i', $guid )) {
+                $guid = str_replace( '-1.pdf', '.pdf', $guid );
+            }*/
+
+            if ( preg_match( '/'.$basename.'$/i', $guid ) ) {
+                $content = str_replace( $media_file, $guid, $content );
+                $update_status = 1;
+            }
+        }
+    }
+
+    if ($update_status === 1) {
+        $updated_post = [];
+        $updated_post["ID"] = $postid;
+        $updated_post["post_content"] = $content;
+        wp_update_post( $updated_post );
+    }
+
+    return [
+        'postid' => $postid ,
+        'update_status' => $update_status,
+    ];
+}
+
+//add_action('pmxi_saved_post','post_saved',5,2);
```

Comment:
I comment this action hook as its more relevant and needed for fresh data import.